logistic_requisition/model/sale_order.py

```diff
@@ -29,6 +30,103 @@ class SaleOrder(models.Model):
                                      ondelete='restrict',
                                      copy=False)
 
+    lr_count = fields.Integer(
+        compute='_count_logistic_requisition'
+    )
+    lr_source_count = fields.Integer(
+        compute='_count_lr_source'
+    )
+    lr_purchase_count = fields.Integer(
+        compute='_count_lr_purchase'
+    )
+
+    @api.multi
+    def action_accepted(self):
+        """ On acceptation of Cost Estimate, we generate PO
+        for all line sourced by a tender
+
+        """
+        res = super(SaleOrder, self).action_accepted()
+        purch_req_so_lines = self.mapped('order_line').filtered(
+            lambda rec: (rec.sourcing_method == 'procurement'
+                         and not rec.sourced_by))
+        todo = self.env['purchase.requisition'].browse()
+        for line in purch_req_so_lines:
+            source = line.lr_source_id
+            if not source.purchase_line_id:
+                todo |= source.po_requisition_id
+
+        for purch_req in todo:
+            purch_req.generate_po()
+
+        for line in purch_req_so_lines:
+            line.sourced_by = line.lr_source_id.purchase_line_id
+        return res
+
+    @api.multi
+    def _get_relevent_purchases(self):
+        purchases = self.env['purchase.order']
+        sourced_lines = self.order_line.filtered(lambda rec: rec.sourced_by)
```

Comment:
replace the lambda func with `"sourced_by"`, filtered handles that case and provides almost the same lambda in that case.